sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -185,6 +185,14 @@ case class FileSourceScanExec(
     ret
   }
 
+  @transient private lazy val selectedPartitionsCount =
+    if (partitionFilters.exists(f =>
+      SubqueryExpression.hasSubquery(f) || ExecSubqueryExpression.hasSubquery(f))) {
```

Comment:
why do we need to check both?

Reply:
Actually `SubqueryExpression.hasSubquery(f)` is not necessary to work correctly. But it was helpful with debugging as `FileSourceScanExec.toString` refers to `metadata` and so throws an exception when `SubqueryExpression` hasn't been transformed to `ExecSubqueryExpression. But I can remove it if having it there is confusing.

Reply:
yes, I'd remove it, thanks.